scripts/script.js

```diff
@@ -0,0 +1,127 @@
+            }
+        }
+    )
+}
+
+function createCityList(list) {
+    for (let i = 0; i < list.length; i++) {
+        const { cityList } = elements
+        const city = createDOMElement('li', 'weather-city-city')
+        city.innerHTML = list[i]
+        city.setAttribute("data-for", list[i])
+        cityList.append(city)
+    }
+}
+
+function toggleList() {
+    const { cityList } = elements
+    cityList.classList.toggle('hidden')
+}
+
+function animation() {
+    const { loading } = elements
+    loading.classList.remove('is-loading')
+}
+
+function selectedCityUpdate() {
+    const { citySelected } = elements
+    citySelected.innerHTML = window.localStorage.getItem('userCity')
+}
+
+async function findCityByName(cityName) {
+    const response = await fetch(`https://api.openweathermap.org/data/2.5/weather?q=${cityName}&appid=${apiKey}&units=metric`)
+    const data = await response.json()
+    if (!response.ok) {
+        responseErrorChecker(data)
+    }
+    updater(data)
+}
+
+async function findCityByGeolocation(lat, lon) {
+    const response = await fetch (`https://api.openweathermap.org/data/2.5/weather?lat=${lat}&lon=${lon}&appid=${apiKey}&units=metric`)
+    const data = await response.json()
+    if (!response.ok) {
+        responseErrorChecker(data)
+    }
+    updater(data)
+}
+
+function updater(data) {
+    const city = new CityInfo(data)
+
+    updateCity(city)
+    selectedCityUpdate()
+}
+
+function updateCity(city) {
+    elements.images.forEach( ele => {
+        const imageElement = document.createElement('img')
+        imageElement.setAttribute('src', `https://openweathermap.org/img/wn/${city.icon}@2x.png`)
+        ele.innerHTML = ""
+        ele.append(imageElement)
+    })
+    city.secondState = firstCharToUpperCase(city.secondState)
+    elements.direction.style.transform = `rotate(${city.windDeg}deg)`
+
+    elements.temperatures.forEach( ele => ele.innerHTML = `${city.temp}° C`)
+    elements.currentState.innerHTML = `${city.secondState}`
+    elements.currentTime.innerHTML = `${city.time}`
+    elements.currentCity.innerHTML = `${city.fullName}`
+    elements.currentInfo.innerHTML = `Feels like ${city.feelsLike}° C. ${city.state}. ${city.secondState}`
+    elements.wind.innerHTML = `${city.windSpeed} m/s ${city.windDirection}`
+    elements.pressure.innerHTML = `${city.pressure} hPa`
+    elements.humidity.innerHTML = `${city.humidity}%`
+    elements.dew.innerHTML = `${city.dew}° C`
+    elements.visibility.innerHTML = `${city.visibility}km`
+
+    window.localStorage.setItem("userCity", city.name)
+}
+
+document.addEventListener('click', (e) => {
+    if (e.target.classList.contains('weather-city-selected')) toggleList();
+    if (e.target.classList.contains('weather-city-city')) findCityByName(e.target.dataset.for);
+})
+
+document.addEventListener("DOMContentLoaded", async () => {
+    userGeolocation()
+    createCityList(cityList)
+    await new Promise(
```

Comment:
delete this inappropriate Promise. I already told you via telegram how to implement preloader behaviour.